scripts/bash_functions/functions.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -e
+WAIT_THRESHOLD="${WAIT_THRESHOLD:-20}"
+SLEEP_SECONDS="${SLEEP_SECONDS:-3}"
+
+
+get_couchdb_url()
+{
+  echo "http://$COUCHDB_USER:$COUCHDB_PASSWORD@$COUCHDB_SERVICE_NAME:$COUCHDB_PORT"
+}
+
+
+wait_for_couchdb()
+{
+  local  wait_count=0
+  while [ "$WAIT_THRESHOLD" -gt $wait_count ]; do
+    wait_count=$((wait_count +1))
+    curl "`get_couchdb_url`" &>/dev/null
+    [ "$?" -eq 0 ] && return 0
+    sleep $SLEEP_SECONDS
+    echo 'Waiting for couchdb'>&2
+  done
+
+  return 1
+}
+
+
+create_couchdb_admin()
+{
+  local user="$1"
+  local url="`get_couchdb_url`"
+
+  # generate password and create user
+  local passwd=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${4:-32};)
+  curl -X PUT $url/_node/couchdb@$COUCH_NODE_NAME/_config/admins/$user -d '"'"$passwd"'"'
```

Comment:
I'm testing in a clustered setup and it seems that I have to add the user to every node in the cluster, otherwise, depending on which node the user gets routed to, they might get on/off success or 401s. 
Can someone please confirm this? 
Thanks!

Reply:
@dianabarsan this was discovered while performing horizontal scalability testing. Check out [this](https://github.com/medic/medic-infrastructure/issues/301#issuecomment-872946524) comment for more descriptions. I had to work around it by making some hack to get it to work.  
Also noting that at the bottom of [the ticket](https://github.com/medic/medic-infrastructure/issues/301) you can see are linked several other tickets that need to be addressed in connection with horizontal scalability, clustering etc.

Reply:
Thanks for confirming, @henokgetachew ! 

Since we're keeping this format of each service creating their own user, we would just create the user on all nodes at each startup. This would imply that, if more nodes are added later, the services would need to restart (and re-create the accounts on all nodes).

Reply:
Correct. As it stands though, the `COUCH_NODE_NAME` is an obstacle for that. Only a single node can be specified using it. The code also configures only that one node with the passwords. This can be avoided if the code queries the `_membership` endpoint to get the list of nodes and configure the admin user on all nodes.

Reply:
Correct. Happy we agree :)

Reply:
What happens when we add a new API container to the fleet, does it create it's own account or it it uses the existing API  containers account?  I think it should create a new account that way we can identify it in the couch logs. That means our user creation logic should consider how API users are named.

Reply:
Yup, `$user` needs some random string appended to it. I think it also makes sense to clean up accounts the pod/container created when it gets killed so we don't have so many orphaned accounts.

Reply:
We're not going to add support for multiple APIs in code in 4.0, so we can delay adding that support in user creation until later as well.